src/processing/ParameterizedOperationsUtil.java

```diff
@@ -0,0 +1,197 @@
+        centrePos.r = (int)(centrePos.r / numOfPixels);
+        centrePos.c = (int)(centrePos.c / numOfPixels);
+
+        return centrePos;
+    }
+
+    /**
+     * Function calculates the rotation matrix.
+     */
+    private static double[][] rotationMatrix(Angle angleOfRotation) {
+
+        // calculating angle, sin, cosine values
+        double radians = Math.toRadians(angleOfRotation.angle);
+        double cosTheta = Math.cos(radians), sinTheta = Math.sin(radians);
+
+        // rotation matrix for counter-clockwise rotation
+        double[][] rotMat = new double[2][2];
+        rotMat[0][0] = cosTheta;
+        rotMat[0][1] = -1.0 * sinTheta;
+        rotMat[1][0] = sinTheta;
+        rotMat[1][1] = cosTheta;
+
+        return rotMat;
+    }
+
+    /**
+     * Function to change colors of object, store all required object's data members in local
+     * variables, call for removing previous object from all utility maps, and finally create
+     * new updated object.
+     * To be called by ClientSideProcessing(vis colorChange function) and UndoRedo sections.
+     */
+    public static BoardObject colorChangeUtil(BoardObject obj, UserId id, Intensity newIntensity) {
+
+        // storing initial intensities of pixels (to be used during undo-operation)
+        ArrayList<Pixel> prevPixelIntensities = new ArrayList<Pixel>();
+        prevPixelIntensities = obj.getPixels();
+
+        // changing intensities of selected object's pixels
+        ArrayList<Pixel> newPixelSet = new ArrayList<Pixel>();
+        newPixelSet = obj.getPixels();
+        for(int i = 0; i < newPixelSet.size(); i++) {
+
+            Pixel newPix = new Pixel(newPixelSet.get(i).position, newIntensity);
+            newPixelSet.set(i, newPix);
+        }
+
+        // storing data members that remain same: objectId, and timestamp of object creation
+        ObjectId newObjectId = obj.getObjectId();
+        Timestamp newTime = obj.getTimestamp();
+
+        // set (COLOR_CHANGE) as the operation which is applied on object.
+        IBoardObjectOperation newBoardOp = new ColorChangeOperation(newIntensity);
+
+        // remove previous object from maps.
+        BoardObject dummyObj = ClientBoardState.maps.removeObjectFromMaps(obj.getObjectId());
+
+        // create a new object with same objectId, timestamp and other updated values.
+        BoardObject newObj = CurveBuilder.drawCurve(newPixelSet, newBoardOp, newObjectId, newTime,
+        id, prevPixelIntensities, false);
+
+        return newObj;
+    }
+
+    /**
+     * Function to rotate the object, store all required object's data members in local
+     * variables, call for removing previous object from all utility maps, and finally create
+     * new updated object.
+     * To be called by ClientSideProcessing(via rotation function) and UndoRedo sections.
+     */
+    public static BoardObject rotationUtil(BoardObject obj, UserId id, Angle angleOfRotation) {
```

Comment:
Like mentioned earlier, "rotationUtil" is not very descriptive. "rotate" or "processRotation" (which indicates there is a bit of processing involved in the rotation functionality) seem clearer to me.

Reply:
@sakshi1603 @RathoreDevansh08 @wildcat26 Did we miss resolving this?